launcher/src/main/java/org/apache/spark/launcher/SparkLauncher.java

```diff
@@ -359,6 +364,82 @@ public SparkLauncher setVerbose(boolean verbose) {
   }
 
   /**
+   * Sets the working directory of the driver process.
+   * @param dir The directory to set as the driver's working directory.
+   * @return This launcher.
+     */
+  public SparkLauncher directory(File dir) {
+    builder.workingDir = dir;
+    return this;
+  }
+
+  /**
+   * Specifies that stderr in the driver should be redirected to stdout.
+   * @return This launcher.
+   */
+  public SparkLauncher redirectError() {
+    builder.redirectErrorStream = true;
+    return this;
+  }
+
+  /**
+   * Redirects error output to the specified Redirect.
+   * @param to The method of redirection.
+   * @return This launcher.
+   */
+  public SparkLauncher redirectError(ProcessBuilder.Redirect to) {
+    builder.errorStream = to;
+    return this;
+  }
+
+  /**
+   * Redirects standard output to the specified Redirect.
+   * @param to The method of redirection.
+   * @return This launcher.
+   */
+  public SparkLauncher redirectOutput(ProcessBuilder.Redirect to) {
+    builder.outputStream = to;
+    return this;
+  }
+
+  /**
+   * Redirects error output to the specified File.
+   * @param errFile The file to which stderr is written.
+   * @return This launcher.
+   */
+  public SparkLauncher redirectError(File errFile) {
+    builder.errorStream = ProcessBuilder.Redirect.to(errFile);
+    return this;
+  }
+
+  /**
+   * Redirects error output to the specified File.
+   * @param outFile The file to which stdout is written.
+   * @return This launcher.
+   */
+  public SparkLauncher redirectOutput(File outFile) {
+    builder.outputStream = ProcessBuilder.Redirect.to(outFile);
+    return this;
+  }
+
+  /**
+   * Sets all output to be logged and redirected to a logger with the specified name.
+   * @param loggerName The name of the logger to log stdout and stderr.
+   * @return This launcher.
+     */
+  public SparkLauncher redirectToLog(String loggerName) {
+    try {
+      // NOTE: the below ordering is important, so builder.redirectToLog is only set to true iff
+      // the preceding put() finishes without exception.
+      builder.getEffectiveConfig().put(CHILD_PROCESS_LOGGER_NAME, loggerName);
```

Comment:
Should I also modify `startApplication` to read from builder.conf? It appears to use `builder.getEffectiveConfig()` which as far as I can tell is sourced from a properties file